clippy_lints/src/instant_elapsed.rs

```diff
@@ -0,0 +1,84 @@
+use clippy_utils::diagnostics::span_lint_and_sugg;
+use rustc_errors::Applicability;
+use rustc_hir::{BinOpKind, BindingAnnotation, Expr, ExprKind, PatKind, StmtKind};
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_session::{declare_tool_lint, impl_lint_pass};
+use rustc_span::source_map::Spanned;
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Catch for substraction between Instant::new() and previous timestamp.
+    ///
+    /// ### Why is this bad?
+    /// This is error prone and there is a specific function for this.
+    ///
+    /// ### Example
+    /// ```rust
+    /// use std::time::Instant;
+    /// let prev_instant = Instant::now();
+    /// let duration = Instant::now() - prev_instant;
+    /// ```
+    /// Use instead:
+    /// ```rust
+    /// use std::time::Instant;
+    /// let prev_instant = Instant::now();
+    /// let duration = prev_instant.elapsed();
+    /// ```
+    #[clippy::version = "1.64.0"]
+    pub INSTANT_ELAPSED,
+    pedantic,
+    "catch substraction between Instant::new() and previous timestamp "
+}
+
+#[derive(Default)]
+pub struct InstantElapsed {
+    variable_name: String,
+    local_id: u32,
+}
+
+impl_lint_pass!(InstantElapsed => [INSTANT_ELAPSED]);
+
+impl LateLintPass<'_> for InstantElapsed {
+    fn check_expr(&mut self, cx: &LateContext<'_>, expr: &'_ Expr<'_>) {
+        if let ExprKind::Block(block, _) = expr.kind {
+            for (_, stmt) in block.stmts.iter().enumerate() {
+                if let StmtKind::Local(local) = stmt.kind
+                    && let PatKind::Binding(BindingAnnotation::Unannotated, local_hir_id, ident, None) = local.pat.kind
+                    && let Some(expr_block) = local.init
+                    && check_instant_now_call(cx, expr_block) {
+                    self.variable_name = ident.to_string();
+                    self.local_id = local_hir_id.local_id.as_u32();
+                }
+
+                if let StmtKind::Local(local) = stmt.kind
+                    && let Some(expr_block) = local.init
+                    && let ExprKind::Binary(Spanned {node: BinOpKind::Sub, ..}, lhs, rhs) = &expr_block.kind
```

Comment:
The local tracking can be removed now it's not used 

```suggestion
        if let ExprKind::Binary(Spanned {node: BinOpKind::Sub, ..}, lhs, rhs) = expr.kind
```

(plus any required balancing braces, formatting)